frontend/src/components/common/form/StepForm/context.tsx

```diff
@@ -0,0 +1,62 @@
+import * as React from 'react';
+import { useFormikContext, getIn } from 'formik';
+import { noop } from 'lodash';
+import { IStepperFormContextProps, IStepperFormProviderProps } from './types';
+
+const StepperFormContext = React.createContext<IStepperFormContextProps>({
+  current: 0,
+  gotoStep: noop,
+  goBack: noop,
+  gotoNext: noop,
+});
+
+export const StepperFormProvider: React.FC<IStepperFormProviderProps> = ({ children, steps }) => {
+  const { values, setFieldValue } = useFormikContext();
+
+  const gotoStep = (index: number) => {
+    if (index > 0 && index < steps.length) {
+      const nextStep = steps[index];
+      if (nextStep.canGoToStep) {
+        setFieldValue('activeStep', index);
+      }
+    }
+  };
+
+  const gotoNext = () => {
```

Comment:
I'd be tempted to return a boolean here to let the consumer now if the next/back action actually occurred.